### PR TITLE
Upgrade services from iceberg 1.2.0 to iceberg 1.5.2

### PR DESCRIPTION
**Pull request by jiang95-dev**
## Summary
Upgrade `services` from iceberg 1.2.0 to iceberg 1.5.2. `Integrations`, `apps`, and `tables-test-fixtures` will remain iceberg 1.2.0.

Notable changes from iceberg 1.2.0 to iceberg 1.5.2 are not limited as following:
- Add view support
- Add FileIO that supports ADLSv2 storage
- Support file and partition delete granularity
- Track partition statistics in TableMetadata
- Add last updated timestamp and snapshot ID to partitions metadata table
- Added support for Spark 3.5 and removed support for Spark 3.2
- Add fast_forward procedure

## Changes
Added `openhouse.iceberg-conventions-1.2` to strictly pin the dependencies version to be 1.2.0. 

Excluded the iceberg libraries from the `tables` and `tables-test-fixtures` module to remove the circular dependency.

**After the change, modules using iceberg 1.5.2 will be**:
- services
- cluster:storage
- internalcatalog
- htscatalog

**Modules using iceberg 1.2.0 will be**:
- apps
- integrations
- libs
- tables-test-fixtures

**Checkboxes**:
- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [ ] Bug Fixes
- [ ] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [x] Refactoring
- [ ] Documentation
- [ ] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

## Testing Done
Comprehensive spark compatibility tests performed in the local cluster: https://docs.google.com/document/d/1yXORH5ety5Gdr6Avsh7XEmIghiJ8wQrg0l4G4_9Soxo/edit#heading=h.h69v5xcbp1md

- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [ ] Added new tests for the changes made.
- [x] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

File: buildSrc/src/main/groovy/openhouse.iceberg-conventions-1.2.gradle [hunk 1]
```diff
@@ -0,0 +1,11 @@
+ext {
+  icebergVersion = '1.2.0'
+}
+
+dependencies {
+  implementation('org.apache.iceberg:iceberg-bundled-guava:' + icebergVersion + "!!")
```
Comment (sumedhsakdeo) [inline buildSrc/src/main/groovy/openhouse.iceberg-conventions-1.2.gradle:6]: Curious what !! do?

File: buildSrc/src/main/groovy/openhouse.iceberg-conventions.gradle [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ext {
-  icebergVersion = '1.2.0'
+  icebergVersion = '1.5.2'
```
Comment (sumedhsakdeo) [inline buildSrc/src/main/groovy/openhouse.iceberg-conventions.gradle:2]: Why is name not iceberg-conventions-1.5.2

File: libs/datalayout/build.gradle [hunk 3]
```diff
@@ -1,7 +1,6 @@
 plugins {
     id 'openhouse.java-conventions'
     id 'openhouse.hadoop-conventions'
-    id 'openhouse.iceberg-conventions'
```
Comment (sumedhsakdeo) [inline libs/datalayout/build.gradle:4]: Why?

File: buildSrc/src/main/groovy/openhouse.iceberg-conventions.gradle (hunk 2 above)
Comment (jiang95-dev) [reply buildSrc/src/main/groovy/openhouse.iceberg-conventions.gradle:2]: Yeah we can change that name too.

File: buildSrc/src/main/groovy/openhouse.iceberg-conventions-1.2.gradle (hunk 1 above)
Comment (jiang95-dev) [reply buildSrc/src/main/groovy/openhouse.iceberg-conventions-1.2.gradle:6]: It forcefully pins the version when there's a conflict.

File: buildSrc/src/main/groovy/openhouse.iceberg-conventions.gradle (hunk 2 above)
Comment (sumedhsakdeo) [reply buildSrc/src/main/groovy/openhouse.iceberg-conventions.gradle:2]: Cool let's do it

**Review by HotSushi — CHANGES_REQUESTED** [review]
Might be looked into already, but is there any spec changes between 1.2 to 1.5? also can we do a staging test before merge?

**Comment by jiang95-dev** [comment]
> Might be looked into already, but is there any spec changes between 1.2 to 1.5? also can we do a staging test before merge?

Added some notable changes, but there's no changes in spec.
Also added the spark compatibility testing report.

File: libs/datalayout/build.gradle (hunk 3 above)
Comment (jiang95-dev) [reply libs/datalayout/build.gradle:4]: Added back.

File: iceberg/openhouse/internalcatalog/src/test/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalTableOperationsTest.java [hunk 4]
```diff
@@ -106,8 +106,9 @@ void testDoCommitAppendSnapshotsInitialVersion() throws IOException {
 
       Map<String, String> updatedProperties = tblMetadataCaptor.getValue().properties();
       Assertions.assertEquals(
-          4,
-          updatedProperties.size()); /*location, lastModifiedTime, version and appended_snapshots*/
+          5,
```
Comment (autumnust) [inline iceberg/openhouse/internalcatalog/src/test/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalTableOperationsTest.java:109]: `write.parquet.compression-codec` why is this showing up as part of iceberg version upgrade?

**Review by autumnust — CHANGES_REQUESTED** [review]
one question. 

Also, when we have things like ADLSv2 storage, can we remove Kai's copy ?

File: iceberg/openhouse/internalcatalog/src/test/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalTableOperationsTest.java (hunk 4 above)
Comment (jiang95-dev) [reply iceberg/openhouse/internalcatalog/src/test/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalTableOperationsTest.java:109]: It is explicitly set by iceberg in 1.5.2: https://github.com/apache/iceberg/blob/5e139f48505b9b622e2cf389eae2bb74d6420510/core/src/main/java/org/apache/iceberg/TableMetadata.java#L93

**Comment by jiang95-dev** [comment]
> one question.
> 
> Also, when we have things like ADLSv2 storage, can we remove Kai's copy ?

I want to do that in a separate PR.

File: apps/spark/build.gradle [hunk 5]
```diff
@@ -1,7 +1,7 @@
 plugins {
     id 'openhouse.java-conventions'
     id 'openhouse.hadoop-conventions'
-    id 'openhouse.iceberg-conventions'
+    id 'openhouse.iceberg-conventions-1.2'
```
Comment (abhisheknath2011) [inline apps/spark/build.gradle:4]: Why this module is using iceberg 1.2?

File: buildSrc/src/main/groovy/openhouse.iceberg-conventions-1.2.gradle [hunk 6]
```diff
@@ -0,0 +1,11 @@
+ext {
+  icebergVersion = '1.2.0'
```
Comment (abhisheknath2011) [inline buildSrc/src/main/groovy/openhouse.iceberg-conventions-1.2.gradle:2]: Is this convention for backward compatibility reason?

File: tables-test-fixtures/build.gradle [hunk 7]
```diff
@@ -2,6 +2,7 @@ plugins {
   id 'java'
   id 'com.github.johnrengelman.shadow' version '7.1.2'
   id 'openhouse.maven-publish'
+  id 'openhouse.iceberg-conventions-1.2'
```
Comment (abhisheknath2011) [inline tables-test-fixtures/build.gradle:5]: Why some modules are on Iceberg 1.2?

File: apps/spark/build.gradle (hunk 5 above)
Comment (jiang95-dev) [reply apps/spark/build.gradle:4]: We are only upgrading the server for now.

File: buildSrc/src/main/groovy/openhouse.iceberg-conventions-1.2.gradle (hunk 6 above)
Comment (jiang95-dev) [reply buildSrc/src/main/groovy/openhouse.iceberg-conventions-1.2.gradle:2]: The plan is upgrade the server and client separately.

**Comment by jiang95-dev** [comment]
Performance the same [tests](https://docs.google.com/document/d/1yXORH5ety5Gdr6Avsh7XEmIghiJ8wQrg0l4G4_9Soxo/edit) in the staging environment. @HotSushi

**Review by HotSushi — APPROVED** [review]

**Review by autumnust — APPROVED** [review]

**Review by abhisheknath2011 — APPROVED** [review]

**Pull request merged**